src/dex/fluid-dex/fluid-dex-pool.ts

```diff
@@ -0,0 +1,243 @@
+import { Interface } from '@ethersproject/abi';
+import { DeepReadonly } from 'ts-essentials';
+import { Log, Logger } from '../../types';
+import { catchParseLogError } from '../../utils';
+import { StatefulEventSubscriber } from '../../stateful-event-subscriber';
+import { IDexHelper } from '../../dex-helper/idex-helper';
+import FluidDexPoolABI from '../../abi/fluid-dex/fluid-dex.abi.json';
+import ResolverABI from '../../abi/fluid-dex/resolver.abi.json';
+import LiquidityABI from '../../abi/fluid-dex/liquidityUserModule.abi.json';
+import { FluidDexPool, FluidDexPoolState, PoolWithReserves } from './types';
+import { ethers } from 'ethers';
+import { eachOfSeries } from 'async';
+import { USD_PRECISION } from '../woo-fi-v2/constants';
+import { MultiResult, MultiCallParams } from '../../lib/multi-wrapper';
+import { BytesLike, defaultAbiCoder } from 'ethers/lib/utils';
+import { Address } from '../../types';
+import { generalDecoder } from '../../lib/decoders';
+
+export class FluidDexEventPool extends StatefulEventSubscriber<FluidDexPoolState> {
+  handlers: {
+    [event: string]: (
+      event: any,
+      state: DeepReadonly<FluidDexPoolState>,
+      log: Readonly<Log>,
+    ) => Promise<DeepReadonly<FluidDexPoolState> | null>;
+  } = {};
+
+  logDecoder: (log: Log) => any;
+
+  addressesSubscribed: Address[];
+  protected liquidityIface = new Interface(LiquidityABI); // TODO: add any additional params required for event subscriber
+
+  constructor(
+    readonly parentName: string,
+    readonly pool: FluidDexPool,
+    protected network: number,
+    protected dexHelper: IDexHelper,
+    logger: Logger,
+  ) {
+    // TODO: Add pool name
+    super(parentName, pool.id, dexHelper, logger);
+
+    // TODO: make logDecoder decode logs that
+    this.logDecoder = (log: Log) => this.liquidityIface.parseLog(log);
+    this.addressesSubscribed = [pool.liquidityUserModule];
```

Comment:
the address that the event is emitted on is the liquidity proxy, not the user module: 0x52Aa899454998Be5b000Ad077a46Bbe360F4e497